libraries/ms-helper/inc/adc_periodic_reader.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "gpio_mcu.h"
+#include "interrupt.h"
+#include "log.h"
+#include "misc.h"
+#include "soft_timer.h"
+#include "status.h"

```

Comment:
Only include headers you use in this file. i.e. type definitions. Similar problem in .c files I think